diff --git a/ext/common/agents/HelperAgent/RequestHandler/ForwardResponse.cpp b/ext/common/agents/HelperAgent/RequestHandler/ForwardResponse.cpp
index 6428408..c9a59cc 100644
--- a/ext/common/agents/HelperAgent/RequestHandler/ForwardResponse.cpp
+++ b/ext/common/agents/HelperAgent/RequestHandler/ForwardResponse.cpp
@@ -714,12 +714,6 @@ constructHeaderBuffersForResponse(Request *req, struct iovec *buffers,
 		PUSH_STATIC_BUFFER("\r\n");
 	}
 
-	if (showVersionInHeader) {
-		PUSH_STATIC_BUFFER("X-Powered-By: " PROGRAM_NAME " " PASSENGER_VERSION "\r\n\r\n");
-	} else {
-		PUSH_STATIC_BUFFER("X-Powered-By: " PROGRAM_NAME "\r\n\r\n");
-	}
-
 	nbuffers = i;
 	return true;
 
diff --git a/ext/common/agents/HelperAgent/RequestHandler/TurboCaching.h b/ext/common/agents/HelperAgent/RequestHandler/TurboCaching.h
index 82ea223..f5cb2a3 100644
--- a/ext/common/agents/HelperAgent/RequestHandler/TurboCaching.h
+++ b/ext/common/agents/HelperAgent/RequestHandler/TurboCaching.h
@@ -154,12 +154,6 @@ private:
 		}
 		PUSH_STATIC_STRING("\r\n");
 
-		if (prep.showVersionInHeader) {
-			PUSH_STATIC_STRING("X-Powered-By: " PROGRAM_NAME " " PASSENGER_VERSION "\r\n");
-		} else {
-			PUSH_STATIC_STRING("X-Powered-By: " PROGRAM_NAME "\r\n");
-		}
-
 		if (server->canKeepAlive(req)) {
 			if (httpVersion < 1010) {
 				// HTTP < 1.1 defaults to "Connection: close"
diff --git a/ext/nginx/ContentHandler.c b/ext/nginx/ContentHandler.c
index 9ba443b..2e400c6 100644
--- a/ext/nginx/ContentHandler.c
+++ b/ext/nginx/ContentHandler.c
@@ -1177,19 +1177,7 @@ process_header(ngx_http_request_t *r)
 
                 h->key.len = sizeof("Server") - 1;
                 h->key.data = (u_char *) "Server";
-                if (!passenger_main_conf.show_version_in_header) {
-                    if (clcf->server_tokens) {
-                        h->value.data = (u_char *) (NGINX_VER " + " PROGRAM_NAME);
-                    } else {
-                        h->value.data = (u_char *) ("nginx + " PROGRAM_NAME);
-                    }
-                } else {
-                    if (clcf->server_tokens) {
-                        h->value.data = (u_char *) (NGINX_VER " + " PROGRAM_NAME " " PASSENGER_VERSION);
-                    } else {
-                        h->value.data = (u_char *) ("nginx + " PROGRAM_NAME " " PASSENGER_VERSION);
-                    }
-                }
+                h->value.data = (u_char *) ("nginx");
                 h->value.len = ngx_strlen(h->value.data);
                 h->lowcase_key = (u_char *) "server";
             }
